sdk/core/Azure.Core/src/Pipeline/DiagnosticsOptions.cs

```diff
@@ -48,7 +52,18 @@ internal DiagnosticsOptions()
         /// </summary>
         public IList<string> LoggingAllowedQueryParameters { get; }
 
-        public string? ApplicationId { get; set; }
+        public string? ApplicationId
```

Comment:
Would it be simpler to use this instead? https://docs.microsoft.com/en-us/dotnet/api/system.configuration.stringvalidatorattribute.maxlength?view=netframework-4.8

Reply:
That's for `System.Configuration` section definitions. It doesn't do any validation by itself.

Reply:
Oh. 😥